### PR TITLE
hyprpm(feat): Support specifying exact git revisions for plugin repo

### LINKED ISSUE #4730: Hyprpm: add ability to specify branch when adding repository
### Description

I might be missing something but this doesn't seem possible currently.

### PR DESCRIPTION
**Pull request by ItsDrike**
#### Describe your PR, what does it fix/add?

This introduces support to specify an exact git revision (commit/tag/branch) for a plugin. The revision can now be specified as the 2nd argument when adding a repo.

When a revision is specified, the commit pins will be ignored, so that the specified rev is used.

Fixes #4730

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

The revision will be specified for the entire repo, not the individual plugins in it. This basically means that all plugins of that repo will use the same locked revision.

#### Is it ready for merging, or does it need work?

Ready.

### CODE REVIEW

**Review by vaxerski — DISMISSED** [review]
fine, but gotta edit the help screen no?

File: hyprpm/src/main.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 const std::string HELP = R"#(┏ hyprpm, a Hyprland Plugin Manager
 ┃
-┣ add [url]              → Install a new plugin repository from git
+┣ add [url] [git rev]    → Install a new plugin repository from git
```
Comment (vaxerski) [inline hyprpm/src/main.cpp:14]: mention its optional, please

Comment (ItsDrike) [reply hyprpm/src/main.cpp:14]: Should I use `<>` to signify optional vars? Or just mention it in the msg?

Comment (vaxerski) [reply hyprpm/src/main.cpp:14]: up to you. If you do <> you need to mention that it means optional ,tho

File: hyprpm/src/main.cpp [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 const std::string HELP = R"#(┏ hyprpm, a Hyprland Plugin Manager
 ┃
-┣ add [url]              → Install a new plugin repository from git
+┣ add [url] [git rev]    → Install a new plugin repository from git. Git revision is optional, when set, commit locks are ignored.
```
Comment (vaxerski) [inline hyprpm/src/main.cpp:14]: come on, you know full well the line is too long. Break it somewhere and align to `Install`

Comment (ItsDrike) [reply hyprpm/src/main.cpp]: was kinda hoping you'd just ignore that lol, sure done

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**